flowmachine/flowmachine/features/subscriber/home_location_monthly.py

```diff
@@ -0,0 +1,197 @@
+from typing import List, Union
+from flowmachine.core.spatial_unit import AnySpatialUnit
+from datetime import datetime, timedelta
+from flowmachine.core.query import Query
+from dateutil.relativedelta import relativedelta
+from dateutil.rrule import rrule, DAILY
+from copy import copy
+
+from flowmachine.features.subscriber.daily_location import daily_location
+from flowmachine.features.subscriber.modal_location import ModalLocation
+from flowmachine.features.subscriber.unique_active_subscribers import (
+    UniqueActiveSubscribers,
+    ActiveSubscribers,
+)
+from flowmachine.features.subscriber.last_location import LastLocation
+from flowmachine.features.subscriber.location_visits import LocationVisits
+
+
+class HomeLocationMonthly(Query):
```

Comment:
Can we make this more generic by saying period instead of month? Understand that our immediate usage context is monthly, but the functionality would equally be applied to fortnights, or biannually, right?

Reply:
Yes, that was my thought too - realised this during implementation.

Reply:
How about `XDaysInYLocation`?

Reply:
How about `ReferencedModalLocation`?